### PR TITLE
Switch to promises

### PR DESCRIPTION
**Pull request by JohnReillyMurray**
switched from callbacks to promises, remove error param since it was always set to false #3

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -73,55 +73,55 @@ module.exports = function (access_token) {
...
+          })
         })
-      })
+      });
     },
 
-    languageSupport: function (callback2) {
-      callback2(languages)
+    languageSupport: function () {
+      return new Promise((resolve, reject) => {
```
Comment (ufocoder) [inline index.js]: Use this shortcut:
```
return Promise.resolve(languages)
```

File: index.js [hunk 2]
```diff
@@ -73,55 +73,55 @@ module.exports = function (access_token) {
...
+              })
+            }else {
+              setTimeout(function () {
+                curl.request(optionsRecv, function (error2, response2) {
+                  var statuscheck = JSON.parse(response2)
+                  if (statuscheck.status == 0) {
+                    curl.request(optionsRecv2, function (error3, response3) {
+                      var answerObject = JSON.parse(response3)
+                      answer = answerObject
+                      resolve(answer);
```
Comment (ufocoder) [inline index.js]: I think this could be refactored
```
resolve(JSON.parse(response3))
```

File: index.js [hunk 3]
```diff
@@ -73,55 +73,55 @@ module.exports = function (access_token) {
...
+            method: 'GET',
+            url: reqUrl
           }
+          curl.request(optionsRecv, function (error2, response2) {
+            var statuscheck = JSON.parse(response2)
+            if (statuscheck.status == 0) {
+              curl.request(optionsRecv2, function (error3, response3) {
+                var answerObject = JSON.parse(response3)
+                answer = answerObject
+                resolve(answer);
```
Comment (ufocoder) [inline index.js]: I think this should be refactored
```
resolve(JSON.parse(response3))
```

File: index.js [hunk 4]
```diff
@@ -73,55 +73,55 @@ module.exports = function (access_token) {
...
-                    var answerObject = JSON.parse(response3)
-                    answer = answerObject
-                    boolResult = false
-                    return callback(answer, boolResult)
-                  })
-                }
-              })
-            }, 4500)
+      return new Promise((resolve, reject) => {
+        curl.request(optionsRun, function (error, response) {
```
Comment (ufocoder) [inline index.js]:
```
if (error) {
  reject(error);
  return;
}
```

File: index.js [hunk 5]
```diff
@@ -73,55 +73,55 @@ module.exports = function (access_token) {
...
+            method: 'GET',
+            url: reqUrl12
+          }
+  
+          var optionsRecv2 = {
+            method: 'GET',
+            url: reqUrl
           }
+          curl.request(optionsRecv, function (error2, response2) {
+            var statuscheck = JSON.parse(response2)
```
Comment (ufocoder) [inline index.js]:
```
if (error2) {
  reject(error2);
  return;
}
```

**Comment by saru95** [comment]
@ufocoder happy to merge the changes if you send a PR :)

**Pull request merged**